cmd_watch.go

```diff
@@ -154,25 +211,33 @@ func evalWorkflow(client *circleci.Client, wfID string, jobName string) (finishe
 
 		switch job.Status {
 		case "success", "blocked":
+			// success means it passed
+			// blocked means it can't yet run, but that could be because either
+			// it's waiting on a running job or
+			// it's not configured to run this build (because of a tag or something)
 			continue
+		case "queued":
+			return false, failed, nil
 		case "failed":
 			failed = true
 			continue
 		case "running":
 			// We can stop short as soon as we find an unfinished job
-			return false, failed
+			return false, failed, nil
 		}
 	}
-	return true, failed
+	return true, failed, nil
 }
 
+// getJobs queries the CircleCI API for a list of all jobs in the current workflow
 func getJobs(client *circleci.Client, wfID string) ([]*circleci.WorkflowJob, error) {
 	// get the list of jobs, paging if necessary
 	wfJobs, more, err := client.ListWorkflowV2Jobs(wfID, nil)
 	if err != nil {
 		return nil, err
 	}
 	for more != nil {
+		// TODO only print this in debug mode
```

Comment:
should this be fixed in this PR or later?

Reply:
there is no debug mode at the moment, and I didn't want to add one in this, since the rest of this is only about watch.